app/src/main/java/ch/epfl/sweng/SDP/home/LeaderboardActivity.java

```diff
@@ -360,20 +372,26 @@ public void onCancelled(@NonNull DatabaseError databaseError) {
         }
 
         /**
-         * Friends button got clicked, now add/remove friend and modify background.
+         * Friends button got clicked, now add/remove friend and modify image.
          *
          * @return listener
          */
-        private ValueEventListener changeFriendsButtonBackgroundOnClick() {
+        private ValueEventListener changeFriendsButtonImageOnClick() {
             return new ValueEventListener() {
                 @Override
                 public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
                     if (dataSnapshot.exists()) {
-                        Account.getInstance(context).removeFriend(player.userId);
-                        setImageResource(R.drawable.add_friend);
+                        int status = (int)(long)dataSnapshot.getValue();
```

Comment:
Here you can do the same trick I told you before

Reply:
Yep, thanks for pointing it out.